lib/components/form/date-time-modal.tsx

```diff
@@ -0,0 +1,117 @@
+// TODO: TypeScript with props.
+/* eslint-disable react/prop-types */
+import { connect } from 'react-redux'
+import coreUtils from '@opentripplanner/core-utils'
+import React, { useCallback } from 'react'
+
+import { AppConfig } from '../../util/config-types'
+import { AppReduxState, FilterType, SortType } from '../../util/state-types'
+import { setQueryParam } from '../../actions/form'
+
+import { StyledDateTimeSelector } from './styled'
+import { updateItineraryFilter } from '../../actions/narrative'
+
+type Props = {
+  config: AppConfig
+  date: string
+  dateFormatLegacy?: string
+  departArrive: DepartArriveValue
+  setQueryParam: (params: any) => void
+  sort: SortType
+  time: string
+  timeFormatLegacy?: string
+  updateItineraryFilter: (payload: FilterType) => void
+}
+
+type DepartArriveValue = 'NOW' | 'DEPART' | 'ARRIVE'
+
+const DepartArriveTypeMap: Record<
+  DepartArriveValue,
+  FilterType['sort']['type']
+> = {
+  ARRIVE: 'ARRIVALTIME',
+  DEPART: 'DEPARTURETIME',
+  NOW: 'DURATION'
+}
+
+function DateTimeModal(props: Props) {
+  const {
+    config,
+    date,
+    dateFormatLegacy,
+    departArrive,
+    setQueryParam,
+    sort,
+    time,
+    timeFormatLegacy,
+    updateItineraryFilter
+  } = props
+  const { homeTimezone, isTouchScreenOnDesktop } = config
+  const touchClassName = isTouchScreenOnDesktop
+    ? 'with-desktop-touchscreen'
+    : ''
+
+  const syncSortWithDepartArrive = config?.itinerary?.syncSortWithDepartArrive
+  const setQueryParamMiddleware = useCallback(
+    (params: any) => {
+      if (syncSortWithDepartArrive) {
+        updateItineraryFilter({
```

Comment:
Maybe worth adding a comment about this side effect: If the user changes the search to "arrive by" and a previous search results are being shown, the current results will be resorted.